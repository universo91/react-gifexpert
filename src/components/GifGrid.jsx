import React from 'react'
import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';
import  PropTypes from 'prop-types';

export const GifGrid =  ({ category }) => {

    // Renombraremos a data como images
    const { loading, data:images } = useFetchGifs( category );
            
    return (
        <>
            <h3 className = "animate__animated animate__fadeIn"> { category } </h3>
            {loading && <p className = "animate__animated animate__flash"> Loading </p> }
            <div className = "card-grid">                
            
                {
                    images.map( img => 
                        <GifGridItem 
                            key = {img.id}
                            {...img } 
                        />
                    )
                }
        
            </div> 
        </>
    )   
    
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}

