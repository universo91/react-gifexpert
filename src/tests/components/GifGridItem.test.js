import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe('Pruebas en GifGriItem', () => {

    const url = 'http://localhost/algo.jpg';
    const title = 'este es un mal dia';

    test('Debe de mostrar correctamente el componente', () => {
        const wrapper = shallow( <GifGridItem url = { url } title = { title } /> );

        expect( wrapper).toMatchSnapshot();
    });

    test('debe tener un parrafo con el title', () => {

        const wrapper = shallow( <GifGridItem title = {title} /> );
        
        const p = wrapper.find('p').text().trim();
        expect( p ).toBe( title );
    });

    test('debe de tener la imagen igual al url y alt de los props', () => {

        const wrapper = shallow( <GifGridItem url = { url } title = { title } />);
        const img = wrapper.find('img');
        

        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );
    });

    test('debe de tener animate_fadeIn', () => {

        const wrapper = shallow( <GifGridItem />);
    
        const div = wrapper.find( 'div' );
        const className = div.prop('className');
    
        expect( className.includes('animate__fadeIn') ).toBe( true );
    });

       
}); 

