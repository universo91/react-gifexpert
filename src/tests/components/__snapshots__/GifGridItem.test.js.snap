// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGriItem Debe de mostrar correctamente el componente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="este es un mal dia"
    src="http://localhost/algo.jpg"
  />
  <p>
    este es un mal dia
     
  </p>
</div>
`;
